File: go/processed_result.go
package main

import (
	"fmt"
	"time"
)

// ProcessedResult represents the result of a processed task
type ProcessedResult struct {
	TaskID           int
	OriginalData     string
	ProcessedData    string
	ProcessingTime   time.Duration
	WorkerThreadName string
	Timestamp        time.Time
}

// NewProcessedResult creates a new processed result
func NewProcessedResult(taskID int, originalData, processedData, workerThreadName string, processingTime time.Duration) *ProcessedResult {
	return &ProcessedResult{
		TaskID:           taskID,
		OriginalData:     originalData,
		ProcessedData:    processedData,
		ProcessingTime:   processingTime,
		WorkerThreadName: workerThreadName,
		Timestamp:        time.Now(),
	}
}

// String returns a string representation of the processed result
func (pr *ProcessedResult) String() string {
	return fmt.Sprintf("ProcessedResult{taskId=%d, originalData='%s', processedData='%s', processingTime=%v, workerThread='%s', timestamp=%s}",
		pr.TaskID, pr.OriginalData, pr.ProcessedData, pr.ProcessingTime, pr.WorkerThreadName, pr.Timestamp.Format("2006-01-02 15:04:05"))
}

File: java/ProcessedResult.java
/**
 * ProcessedResult class representing the result of a processed task
 */
public class ProcessedResult {
    private int taskId;
    private String originalData;
    private String processedData;
    private long processingTime;
    private String workerThreadName;
    private long timestamp;
    
    public ProcessedResult(int taskId, String originalData, String processedData, 
                          long processingTime, String workerThreadName) {
        this.taskId = taskId;
        this.originalData = originalData;
        this.processedData = processedData;
        this.processingTime = processingTime;
        this.workerThreadName = workerThreadName;
        this.timestamp = System.currentTimeMillis();
    }
    
    public int getTaskId() {
        return taskId;
    }
    
    public String getOriginalData() {
        return originalData;
    }
    
    public String getProcessedData() {
        return processedData;
    }
    
    public long getProcessingTime() {
        return processingTime;
    }
    
    public String getWorkerThreadName() {
        return workerThreadName;
    }
    
    public long getTimestamp() {
        return timestamp;
    }
    
    @Override
    public String toString() {
        return "ProcessedResult{taskId=" + taskId + 
               ", originalData='" + originalData + "'" +
               ", processedData='" + processedData + "'" +
               ", processingTime=" + processingTime + "ms" +
               ", workerThread='" + workerThreadName + "'" +
               ", timestamp=" + timestamp + "}";
    }
}
